packages/syft/src/syft/service/project/project_stash.py

```diff
@@ -40,11 +40,17 @@ def get_all_for_verify_key(
         return self.query_all(
             credentials=credentials,
             qks=qks,
-            order_by=OrderByNamePartitionKey,
+            order_by=NamePartitionKey,
         )
 
     def get_by_uid(
         self, credentials: SyftVerifyKey, uid: UID
     ) -> Result[Optional[Project], str]:
         qks = QueryKeys(qks=[UIDPartitionKey.with_obj(uid)])
         return self.query_one(credentials=credentials, qks=qks)
+
+    def get_by_name(
+        self, credentials: SyftVerifyKey, project_name: str
+    ) -> Result[Optional[Project], str]:
+        qks = QueryKeys(qks=[NamePartitionKey.with_obj(project_name)])
+        return self.query_one(credentials=credentials, qks=qks)
```

Comment:
New API call to get directly by name instead of iterating-and-filter over all projects. `"name"` column is now unique, and hence makes sure that it will return only one item

Reply:
Wonderful